test/testsymboldatabase.cpp

```diff
@@ -5803,6 +5804,25 @@ class TestSymbolDatabase : public TestFixture {
         ASSERT_EQUALS(f->valueType()->type, ValueType::Type::INT);
     }
 
+    void enum15() {
+        GET_SYMBOL_DB("struct S {\n"
+                      "    S();\n"
+                      "    enum E { E0 };\n"
+                      "};\n"
+                      "S::S() {\n"
+                      "    E e = E::E0;\n"
```

Comment:
@chrchr-github sorry for late review

I fear that we don't handle inner scopes well:

        GET_SYMBOL_DB("struct S {\n"
                      "    S();\n"
                      "    enum E { E0 };\n"
                      "};\n"
                      "S::S() {\n"
                      "    if (x) { E e = E::E0; }\n"
                      "}\n");

The condition you added `scope->functionOf && ` will be false right?

Reply:
I tried to improve it here: https://github.com/danmar/cppcheck/pull/5459